modules/renter/upload.go

```diff
@@ -102,6 +103,10 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 	lockID := r.mu.Lock()
 	defer r.mu.Unlock(lockID)
 
+	if filepath.Ext(up.Filename) != filepath.Ext(up.Nickname) {
+		return errors.New("nickname and file name must have the same extension")
+	}
```

Comment:
that is a complicated fix sadly.

It's important for sharing that the file extension be reasonable, at least at the moment. I think this is an okay solution.

Auto-loading the name is going to be a good feature to have, but I'm putting it off for the time being.